src/Microsoft.Health.Fhir.Liquid.Converter/DotLiquids/TemplateLocalFileSystem.cs

```diff
@@ -91,22 +90,22 @@ private string GetAbsoluteTemplatePath(string templateName)
             var result = _templateDirectory;
```

Comment:
Is it possible to simplify logics of `GetAbsoluteTemplatePath`? Seems that all the circumstances follow the same handling logic:

1. Update the last element of `pathSegments` based on the type:
    1. If code mapping template: update `pathSegments[^1]` -> `pathSegments[^1].json`
    2. If non root template and non JSON schema: update `pathSegments[^1]` -> `_pathSegments[^1].liquid`
    3. Other conditions just keep as is
4. Aggregate the `pathSegments` with `result` and return.